vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/video/ScreenRecorderTestBatchListener.kt

```diff
@@ -35,54 +36,78 @@ class ScreenRecorderTestBatchListener(
     coroutineScope: CoroutineScope
 ) : NoOpTestRunListener(), AttachmentProvider, CoroutineScope by coroutineScope {
 
-    val attachmentListeners = mutableListOf<AttachmentListener>()
+    private val attachmentListeners = mutableListOf<AttachmentListener>()
 
     override fun registerListener(listener: AttachmentListener) {
         attachmentListeners.add(listener)
     }
 
     private val logger = MarathonLogging.logger("ScreenRecorder")
 
-    private val screenRecorderStopper = ScreenRecorderStopper(device)
+    private val screenRecorder = ScreenRecorder(device, videoConfiguration)
 
     private var hasFailed: Boolean = false
     private var supervisorJob: Job? = null
-    private var lastRemoteFile: String? = null
+    private var lastRemoteFiles: MutableList<String> = mutableListOf()
+    private val chunkedRecording = device.apiLevel < 34
+    private var chunkMap: HashMap<TestIdentifier, Long> = HashMap<TestIdentifier, Long>().apply { withDefault { 0 } }
+    @Volatile
+    private var cancelling: Boolean = false
```

Comment:
`@volatile` is enough? Maybe `AtomicBoolean`?

Reply:
It should be enough for the purpose of synchronizing the stop of the screen recorder